### PR TITLE
Feature/spear bits audit3

### PR DESCRIPTION
**Pull request by invocamanman**
This PR solves the following issues:
https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/18#issuecomment-1403372039
https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/14
https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/16
https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/17
https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/18
https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/20
https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/21
https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/25
https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/26
https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/30
https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/31
https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/33
https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/36

### CODE REVIEW

File: contracts/PolygonZkEVM.sol [hunk 1]
```diff
@@ -660,6 +689,7 @@ contract PolygonZkEVM is OwnableUpgradeable, EmergencyManager {
 
     /**
      * @notice Verify batches internal function
```
Comment (krlosMata) [inline contracts/PolygonZkEVM.sol:691]:
```suggestion
     * @notice Verify and reward batches internal function
```

File: contracts/PolygonZkEVMBridge.sol [hunk 2]
```diff
@@ -153,6 +155,9 @@ contract PolygonZkEVMBridge is
             // Ether is treated as ether from mainnet
             originNetwork = MAINNET_NETWORK_ID;
         } else {
+            // Check msg.value is 0 if tokens are bridged
+            require(msg.value == 0, "PolygonZkEVMBridge::bridgeAsset: Expected zero native asset value when bridging ERC20 tokens");
```
Comment (krlosMata) [inline contracts/PolygonZkEVMBridge.sol:159]: What if there is an ERC20 token that requires some `msg.value` in order to perform its `safeTransferFrom` custom implementation?

**Review by krlosMata — CHANGES_REQUESTED** [review]

File: contracts/PolygonZkEVMBridge.sol (hunk 2 above)
Comment (invocamanman) [reply contracts/PolygonZkEVMBridge.sol:159]: This is not contemplated, and if it was the case we should send the msg.value along with the transferFrom function, which i think by usually is not payable

Comment (krlosMata) [reply contracts/PolygonZkEVMBridge.sol:160]: This avoids bad usage of the Bridge but I think it is not required xD

**Review by krlosMata — APPROVED** [review]
:smile_cat:

**Review by ignasirv — APPROVED** [review]
💯

**Pull request merged**